File: src/main/java/particle/jobjects/Link.java
package particle.jobjects;

import static particle.jobjects.Settings.LINK_FORCE;

public class Link {

    private Particle a;
    private Particle b;

    public Link(Particle a, Particle b) {
        this.a = a;
        this.b = b;
        doLink();
    }

    private void doLink() {
        a.linkWith(b);
    }

    void unlink() {
        a.unlinkFrom(b);
    }

    float squaredDistance() {
        return a.squaredDistanceTo(b);
    }

    void adjustParticlesVelocity() {
        double angle = a.angleTo(b);
        a.addVelocityToPositiveDirection(angle, LINK_FORCE);
        b.addVelocityToNegativeDirection(angle, LINK_FORCE);
    }

    int screenX1() {
        return a.screenX();
    }

    int screenY1() {
        return a.screenY();
    }

    int screenX2() {
        return b.screenX();
    }

    int screenY2() {
        return b.screenY();
    }
}

File: src/main/kotlin/particle/kobjects/Link.kt
package particle.kobjects

import particle.kobjects.Settings.LINK_FORCE


class Link(private val a: Particle, private val b: Particle) {

    init {
        doLink()
    }

    private fun doLink() {
        a.linkWith(b)
    }

    internal fun unlink() {
        a.unlinkFrom(b)
    }

    internal fun squaredDistance(): Float {
        return a.squaredDistanceTo(b)
    }

    internal fun adjustParticlesVelocity() {
        val angle = a.angleTo(b)
        a.addVelocityToPositiveDirection(angle, LINK_FORCE)
        b.addVelocityToNegativeDirection(angle, LINK_FORCE)
    }

    internal fun screenX1(): Int {
        return a.screenX
    }

    internal fun screenY1(): Int {
        return a.screenY
    }

    internal fun screenX2(): Int {
        return b.screenX
    }

    internal fun screenY2(): Int {
        return b.screenY
    }
}